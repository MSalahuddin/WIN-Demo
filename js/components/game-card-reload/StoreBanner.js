import React from 'react';
import styled from 'styled-components/native';
import PropTypes from 'prop-types';

import { Platform } from 'react-native';
import { scale, scaleHeight } from '../../platformUtils';
import Text, { FONT_FAMILY, SIZE } from '../common/Text';
import Arrow, { ARROW_DIRECTION } from '../common/Arrow';
import { color } from '../../styles';
import { BlankBanner, StarBanner } from '../../../assets/images';

export const BANNER_TYPE = {
  BACKDROP: 'BACKDROP',
  BOW_TIE: 'BOW_TIE',
  NORMAL: 'NORMAL',
  STAR: 'STAR',
  PURPLE: 'PURPLE'
};

const getHeight = (bannerType, width) => {
  switch (bannerType) {
    case BANNER_TYPE.BOW_TIE:
      return scale(width) * (85 / 345);
    case BANNER_TYPE.BACKDROP:
      return scale(width) * (66 / 345);
    default:
      return width * (66 / 345);
  }
};

const VIPBannerContainer = styled.View`
  align-items: center;
  justify-content: center;
  margin-top:${Platform.OS === "android" ? 0 : 12}
`;

const VIPBanner = Platform.OS === "android" ?
  styled.Image`
  height: ${({ bannerType, width }) => getHeight(bannerType, width)};
  width: ${({ width }) => width};
  elevation: 3;
   z-index: 2;
`
  : styled.Image`
  height: ${({ bannerType, width }) => getHeight(bannerType, width)};
  width: ${({ width }) => width};
`;

const BannerText = Platform.OS === 'android' ? styled(Text)`
position: absolute;
color: ${color.white};
elevation: 4;
z-index: 3;
padding-bottom: ${scaleHeight(12)};
${({ bannerType }) => bannerType === BANNER_TYPE.BOW_TIE && `padding-bottom: ${scaleHeight(20)}`};
letter-spacing: 1.5;
`:
  styled(Text)`
  position: absolute;
  padding-bottom: ${scaleHeight(5)};
  color: ${color.white};
  ${({ bannerType }) => bannerType === BANNER_TYPE.BOW_TIE && `padding-top: ${scaleHeight(20)}`};
  letter-spacing: 1.5;
`;

// top position should be the height of the image and minus 10
// to move it to the top a bit.
const Backdrop = Platform.OS === 'android' ? styled.View`
background-color: ${color.white};
width: ${scale(50)}%;
border-bottom-left-radius: ${scaleHeight(8)};
border-bottom-right-radius: ${scaleHeight(8)};
height: ${scaleHeight(49)};
justify-content: space-between;
flex-direction: row;
opacity: 0.85;
top: ${scaleHeight(49)};
padding-top: ${scaleHeight(15)};
position: absolute;
shadow-color: ${color.blackShadow};
shadow-offset: 4px 4px;
shadow-opacity: ${1};
shadow-radius: 2px;
elevation: 2;
z-index: 1;
`:
  styled.View`
align-items: center;
background-color: ${color.white};
border-bottom-left-radius: ${scaleHeight(8)};
border-bottom-right-radius: ${scaleHeight(8)};
bottom: ${-scaleHeight(22)};
height: ${scaleHeight(49)};
justify-content: space-between;
flex-direction: row;
opacity: 0.85;
padding-top: ${scaleHeight(15)};
position: absolute;
shadow-color: ${color.blackShadow};
shadow-offset: 4px 4px;
shadow-opacity: ${1};
shadow-radius: 2px;
width: ${scale(50)}%;
elevation: 8;
z-index: -1;
`;

const BackdropText = Platform.OS === 'android' ? styled(Text)`
margin-top: ${scaleHeight(0)};
`:
  styled(Text)`
  margin-top: ${scaleHeight(6)};
`;

const StoreBanner = ({
  label,
  bannerType,
  backdropText,
  textSize,
  width,
  leftAction,
  rightAction,
  leftEnabled,
  rightEnabled,
  ...rest
}) => {
  const renderBackdrop = () => {
    return (
      <Backdrop width={width}>
        <Arrow
          testID="arrow-left"
          arrowDirection={ARROW_DIRECTION.LEFT}
          onPress={() => {
            if (leftEnabled) {
              leftAction();
            }
          }}
          color={leftEnabled ? color.black : color.lightGrey}
        />
        <BackdropText size={SIZE.XSMALL} fontFamily={FONT_FAMILY.SEMI_BOLD}>
          {backdropText}
        </BackdropText>
        <Arrow
          testID="arrow-right"
          arrowDirection={ARROW_DIRECTION.RIGHT}
          onPress={() => {
            if (rightEnabled) {
              rightAction();
            }
          }}
          color={rightEnabled ? color.black : color.lightGrey}
        />
      </Backdrop>
    );
  };

  return (
    <VIPBannerContainer width={width} {...rest}>
      <VIPBanner
        width={width}
        source={bannerType === BANNER_TYPE.STAR ? StarBanner : BlankBanner}
        resizeMode="contain"
        bannerType={bannerType} />
      <BannerText allowFontScaling={false} fontFamily={FONT_FAMILY.BOLD}
        size={textSize} isUppercase bannerType={bannerType}>
        {label}
      </BannerText>
      {bannerType === BANNER_TYPE.BACKDROP && renderBackdrop()}
    </VIPBannerContainer>
  );
};

StoreBanner.propTypes = {
  backdropText: PropTypes.string,
  bannerType: PropTypes.oneOf([
    BANNER_TYPE.NORMAL,
    BANNER_TYPE.BOW_TIE,
    BANNER_TYPE.BACKDROP,
    BANNER_TYPE.STAR,
    BANNER_TYPE.PURPLE
  ]),
  leftEnabled: PropTypes.bool,
  rightEnabled: PropTypes.bool,
  leftAction: PropTypes.func,
  rightAction: PropTypes.func,
  label: PropTypes.string.isRequired,
  width: PropTypes.number.isRequired,
  textSize: PropTypes.oneOf([
    SIZE.NORMAL,
    SIZE.LARGE,
    SIZE.BANNER_LARGE,
    SIZE.XLARGE,
    SIZE.XXLARGE,
    SIZE.LARGEST,
    SIZE.SMALL,
    SIZE.XSMALL,
    SIZE.XXSMALL
  ])
};

StoreBanner.defaultProps = {
  backdropText: '',
  bannerType: BANNER_TYPE.NORMAL,
  textSize: SIZE.XSMALL,
  leftEnabled: true,
  rightEnabled: true,
  leftAction: () => { },
  rightAction: () => { }
};

export default StoreBanner;
