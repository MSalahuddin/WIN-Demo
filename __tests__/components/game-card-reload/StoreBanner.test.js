import React from 'react';
import { render, fireEvent } from 'react-native-testing-library';

import StoreBanner from '../../../js/components/game-card-reload/StoreBanner';

describe('Component StoreBanner', () => {
  const leftAction = jest.fn();
  const rightAction = jest.fn();

  test('renders correctly', () => {
    const output = render(<StoreBanner label="Foo" width={100} textSize="LARGE" bannerType="BACKDROP" />);
    expect(output).toMatchSnapshot();
  });

  test('renders correctly with leftEnabled set to false', () => {
    const output = render(
      <StoreBanner leftEnabled={false} label="Foo" width={100} textSize="LARGE" bannerType="BACKDROP" />
    );
    expect(output).toMatchSnapshot();
  });

  test('renders correctly with rightEnabled set to false', () => {
    const output = render(
      <StoreBanner rightEnabled={false} label="Foo" width={100} textSize="LARGE" bannerType="BACKDROP" />
    );
    expect(output).toMatchSnapshot();
  });

  test('calls leftAction when left arrow is pressed', () => {
    const { getByTestId } = render(
      <StoreBanner label="Foo" width={100} textSize="LARGE" bannerType="BACKDROP" leftAction={leftAction} />
    );
    fireEvent(getByTestId('arrow-left'), 'press');
    expect(leftAction).toHaveBeenCalled();
  });

  test('calls rightAction when right arrow is pressed', () => {
    const { getByTestId } = render(
      <StoreBanner label="Foo" width={100} textSize="LARGE" bannerType="BACKDROP" rightAction={rightAction} />
    );
    fireEvent(getByTestId('arrow-right'), 'press');
    expect(rightAction).toHaveBeenCalled();
  });
});
