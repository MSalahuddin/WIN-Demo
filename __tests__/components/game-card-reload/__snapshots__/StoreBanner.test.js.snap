// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component StoreBanner renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
  width={100}
>
  <Image
    bannerType="BACKDROP"
    resizeMode="contain"
    source={1}
    style={
      Array [
        Object {
          "height": 28.695652173913043,
          "width": 100,
        },
      ]
    }
    width={100}
  />
  <Text
    alignCenter={false}
    allowFontScaling={false}
    bannerType="BACKDROP"
    color="#000000"
    fontFamily="BOLD"
    isUnderlined={false}
    isUppercase={true}
    maxFontSizeMultiplier={1.3}
    size="LARGE"
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "CalibreBold",
          "fontSize": 31.5,
          "justifyContent": "center",
          "textAlign": "left",
          "textDecorationColor": "#000000",
          "textDecorationLine": "none",
          "textDecorationStyle": "solid",
          "textTransform": "uppercase",
        },
        Object {
          "color": "#FFFFFF",
          "letterSpacing": 1.5,
          "paddingBottom": 10,
          "position": "absolute",
        },
      ]
    }
  >
    Foo
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderBottomLeftRadius": 16,
          "borderBottomRightRadius": 16,
          "bottom": -44,
          "elevation": 8,
          "flexDirection": "row",
          "height": 98,
          "justifyContent": "space-between",
          "opacity": 0.85,
          "paddingTop": 30,
          "position": "absolute",
          "shadowColor": "rgba(53, 53, 53, 0.3)",
          "shadowOffset": Object {
            "height": 4,
            "width": 4,
          },
          "shadowOpacity": 1,
          "shadowRadius": 2,
          "width": "75%",
          "zIndex": -1,
        },
      ]
    }
    width={100}
  >
    <View
      accessible={true}
      focusable={true}
      hitSlop={
        Object {
          "bottom": 20,
          "left": 20,
          "right": 20,
          "top": 20,
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="arrow"
    >
      <View
        arrowDirection="LEFT"
        color="#000000"
        size={8}
        style={
          Array [
            Object {
              "borderBottomColor": "transparent",
              "borderLeftColor": "transparent",
              "borderRadius": 3,
              "borderRightColor": "#000000",
              "borderStyle": "solid",
              "borderTopColor": "transparent",
              "borderWidth": 12,
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
    </View>
    <Text
      alignCenter={false}
      allowFontScaling={true}
      color="#000000"
      fontFamily="SEMI BOLD"
      isUnderlined={false}
      isUppercase={false}
      maxFontSizeMultiplier={1.3}
      size="XSMALL"
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "Calibre-Semibold",
            "fontSize": 24,
            "justifyContent": "center",
            "textAlign": "left",
            "textDecorationColor": "#000000",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
            "textTransform": "none",
          },
          Object {
            "marginTop": 12,
          },
        ]
      }
    >
      
    </Text>
    <View
      accessible={true}
      focusable={true}
      hitSlop={
        Object {
          "bottom": 20,
          "left": 20,
          "right": 20,
          "top": 20,
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="arrow"
    >
      <View
        arrowDirection="RIGHT"
        color="#000000"
        size={8}
        style={
          Array [
            Object {
              "borderBottomColor": "transparent",
              "borderLeftColor": "#000000",
              "borderRadius": 3,
              "borderRightColor": "transparent",
              "borderStyle": "solid",
              "borderTopColor": "transparent",
              "borderWidth": 12,
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`Component StoreBanner renders correctly with leftEnabled set to false 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
  width={100}
>
  <Image
    bannerType="BACKDROP"
    resizeMode="contain"
    source={1}
    style={
      Array [
        Object {
          "height": 28.695652173913043,
          "width": 100,
        },
      ]
    }
    width={100}
  />
  <Text
    alignCenter={false}
    allowFontScaling={false}
    bannerType="BACKDROP"
    color="#000000"
    fontFamily="BOLD"
    isUnderlined={false}
    isUppercase={true}
    maxFontSizeMultiplier={1.3}
    size="LARGE"
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "CalibreBold",
          "fontSize": 31.5,
          "justifyContent": "center",
          "textAlign": "left",
          "textDecorationColor": "#000000",
          "textDecorationLine": "none",
          "textDecorationStyle": "solid",
          "textTransform": "uppercase",
        },
        Object {
          "color": "#FFFFFF",
          "letterSpacing": 1.5,
          "paddingBottom": 10,
          "position": "absolute",
        },
      ]
    }
  >
    Foo
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderBottomLeftRadius": 16,
          "borderBottomRightRadius": 16,
          "bottom": -44,
          "elevation": 8,
          "flexDirection": "row",
          "height": 98,
          "justifyContent": "space-between",
          "opacity": 0.85,
          "paddingTop": 30,
          "position": "absolute",
          "shadowColor": "rgba(53, 53, 53, 0.3)",
          "shadowOffset": Object {
            "height": 4,
            "width": 4,
          },
          "shadowOpacity": 1,
          "shadowRadius": 2,
          "width": "75%",
          "zIndex": -1,
        },
      ]
    }
    width={100}
  >
    <View
      accessible={true}
      focusable={true}
      hitSlop={
        Object {
          "bottom": 20,
          "left": 20,
          "right": 20,
          "top": 20,
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="arrow"
    >
      <View
        arrowDirection="LEFT"
        color="#B7B7B7"
        size={8}
        style={
          Array [
            Object {
              "borderBottomColor": "transparent",
              "borderLeftColor": "transparent",
              "borderRadius": 3,
              "borderRightColor": "#B7B7B7",
              "borderStyle": "solid",
              "borderTopColor": "transparent",
              "borderWidth": 12,
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
    </View>
    <Text
      alignCenter={false}
      allowFontScaling={true}
      color="#000000"
      fontFamily="SEMI BOLD"
      isUnderlined={false}
      isUppercase={false}
      maxFontSizeMultiplier={1.3}
      size="XSMALL"
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "Calibre-Semibold",
            "fontSize": 24,
            "justifyContent": "center",
            "textAlign": "left",
            "textDecorationColor": "#000000",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
            "textTransform": "none",
          },
          Object {
            "marginTop": 12,
          },
        ]
      }
    >
      
    </Text>
    <View
      accessible={true}
      focusable={true}
      hitSlop={
        Object {
          "bottom": 20,
          "left": 20,
          "right": 20,
          "top": 20,
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="arrow"
    >
      <View
        arrowDirection="RIGHT"
        color="#000000"
        size={8}
        style={
          Array [
            Object {
              "borderBottomColor": "transparent",
              "borderLeftColor": "#000000",
              "borderRadius": 3,
              "borderRightColor": "transparent",
              "borderStyle": "solid",
              "borderTopColor": "transparent",
              "borderWidth": 12,
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`Component StoreBanner renders correctly with rightEnabled set to false 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
  width={100}
>
  <Image
    bannerType="BACKDROP"
    resizeMode="contain"
    source={1}
    style={
      Array [
        Object {
          "height": 28.695652173913043,
          "width": 100,
        },
      ]
    }
    width={100}
  />
  <Text
    alignCenter={false}
    allowFontScaling={false}
    bannerType="BACKDROP"
    color="#000000"
    fontFamily="BOLD"
    isUnderlined={false}
    isUppercase={true}
    maxFontSizeMultiplier={1.3}
    size="LARGE"
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "CalibreBold",
          "fontSize": 31.5,
          "justifyContent": "center",
          "textAlign": "left",
          "textDecorationColor": "#000000",
          "textDecorationLine": "none",
          "textDecorationStyle": "solid",
          "textTransform": "uppercase",
        },
        Object {
          "color": "#FFFFFF",
          "letterSpacing": 1.5,
          "paddingBottom": 10,
          "position": "absolute",
        },
      ]
    }
  >
    Foo
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderBottomLeftRadius": 16,
          "borderBottomRightRadius": 16,
          "bottom": -44,
          "elevation": 8,
          "flexDirection": "row",
          "height": 98,
          "justifyContent": "space-between",
          "opacity": 0.85,
          "paddingTop": 30,
          "position": "absolute",
          "shadowColor": "rgba(53, 53, 53, 0.3)",
          "shadowOffset": Object {
            "height": 4,
            "width": 4,
          },
          "shadowOpacity": 1,
          "shadowRadius": 2,
          "width": "75%",
          "zIndex": -1,
        },
      ]
    }
    width={100}
  >
    <View
      accessible={true}
      focusable={true}
      hitSlop={
        Object {
          "bottom": 20,
          "left": 20,
          "right": 20,
          "top": 20,
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="arrow"
    >
      <View
        arrowDirection="LEFT"
        color="#000000"
        size={8}
        style={
          Array [
            Object {
              "borderBottomColor": "transparent",
              "borderLeftColor": "transparent",
              "borderRadius": 3,
              "borderRightColor": "#000000",
              "borderStyle": "solid",
              "borderTopColor": "transparent",
              "borderWidth": 12,
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
    </View>
    <Text
      alignCenter={false}
      allowFontScaling={true}
      color="#000000"
      fontFamily="SEMI BOLD"
      isUnderlined={false}
      isUppercase={false}
      maxFontSizeMultiplier={1.3}
      size="XSMALL"
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "Calibre-Semibold",
            "fontSize": 24,
            "justifyContent": "center",
            "textAlign": "left",
            "textDecorationColor": "#000000",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
            "textTransform": "none",
          },
          Object {
            "marginTop": 12,
          },
        ]
      }
    >
      
    </Text>
    <View
      accessible={true}
      focusable={true}
      hitSlop={
        Object {
          "bottom": 20,
          "left": 20,
          "right": 20,
          "top": 20,
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="arrow"
    >
      <View
        arrowDirection="RIGHT"
        color="#B7B7B7"
        size={8}
        style={
          Array [
            Object {
              "borderBottomColor": "transparent",
              "borderLeftColor": "#B7B7B7",
              "borderRadius": 3,
              "borderRightColor": "transparent",
              "borderStyle": "solid",
              "borderTopColor": "transparent",
              "borderWidth": 12,
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
